src/libs/actions/Welcome.ts

```diff
@@ -124,6 +125,17 @@ function checkTryNewDotDataReady() {
     resolveTryNewDotStatus?.();
 }
 
+/**
+ * Check if the onboarding data is loaded
+ */
+function checkOnboardingDataReady() {
+    if (onboarding === undefined) {
+        return;
+    }
```

Comment:
Because we early return here, anonymous account with no on boarding data will stuck at home screen because [this promise](https://github.com/Expensify/App/blame/7b9e55ed202ce242d70b1e8ab0299d2d3fbdd774/src/libs/actions/Welcome/index.ts#L45-L50) won't be executed.
More details:
https://github.com/Expensify/App/issues/55239#issuecomment-2592167692